src/Bicep.Types.UnitTests/TypeSerializerTests.cs

```diff
@@ -95,7 +96,10 @@ public void Different_types_can_be_serialized_and_deserialized()
             #pragma warning restore 618
             var objectType = factory.Create(() => new ObjectType("steven", new Dictionary<string, ObjectTypeProperty>(), null));
             var arrayType = factory.Create(() => new ArrayType(factory.GetReference(objectType)));
-            var resourceType = factory.Create(() => new ResourceType("gerrard", ScopeType.ResourceGroup|ScopeType.Tenant, ScopeType.Tenant, factory.GetReference(objectType), ResourceFlags.None));
+            var stringType = factory.Create(() => new StringType(true, 3, 10, "^foo"));
+            var functionParam = new FunctionTypeParameter("arg", factory.GetReference(stringType), null);
+            var resourceMethodType = factory.Create(() => new FunctionType("sayHi", factory.GetReference(stringType), new [] { functionParam }, null));
+            var resourceType = factory.Create(() => new ResourceType("gerrard", ScopeType.ResourceGroup|ScopeType.Tenant, ScopeType.Tenant, factory.GetReference(objectType), ResourceFlags.None, new [] { factory.GetReference(resourceMethodType) }));
```

Comment:
What should the scope be for a non-Azure type?

Reply:
I think `ScopeType.Unknown` may work, but I'm not 100%. I'm not making any changes to this logic here.